### PR TITLE
Adds more broadcast interfaces to ParallelEnv.f90

### PR DESCRIPTION
**Pull request by aarograh**

### CODE REVIEW

File: src/ParallelEnv.f90 [hunk 1]
```diff
@@ -152,46 +152,62 @@ SUBROUTINE ParEnvType_clear_absintfc(myPE)
...
-                          bcast_SSK4_MPI_Env_type, &
                           bcast_SDK4_MPI_Env_type
       !> @copybrief ParallelEnv::allReduceR_MPI_Env_type
       !> @copydetails  ParallelEnv::allReduceR_MPI_Env_type
       PROCEDURE,PASS :: allReduce => allReduceR_MPI_Env_type
       !> @copybrief ParallelEnv::allReduceI_MPI_Env_type
       !> @copydetails  ParallelEnv::allReduceI_MPI_Env_type
       PROCEDURE,PASS :: allReduceI => allReduceI_MPI_Env_type
-      !GENERIC :: allReduce => allReduceR_MPI_Env_type,allReduceI_MPI_Env_type
+      !GENERIC :: allReduce => allReduceI_MPI_Env_type,allReduceI_MPI_Env_type
```
Comment (bscollin) [inline src/ParallelEnv.f90]: What's going on here?  You got rid of ```allReduceR``` and added two ```AllReduceI```'s?

File: src/ParallelEnv.f90 [hunk 2]
```diff
@@ -739,6 +755,46 @@ SUBROUTINE scatter_SLK1_MPI_Env_type(myPE,sendbuf,recvbuf,root)
...
+!> @brief
+    SUBROUTINE bcast_SNK0_MPI_Env_type(myPE,buf,root)
+      CHARACTER(LEN=*),PARAMETER :: myName='bcast_SNK0_MPI_Env_type'
+      CLASS(MPI_EnvType),INTENT(INOUT) :: myPE
+      INTEGER(SNK),INTENT(IN) :: buf
+      INTEGER(SIK),INTENT(IN),OPTIONAL :: root
+      INTEGER(SIK) :: rank
+      rank=0
+      IF(PRESENT(root)) rank=root
+      IF(0 <= rank .AND. rank <= myPE%nproc-1) THEN
```
Comment (bscollin) [inline src/ParallelEnv.f90]: Seems like a prime place for an assertion.

File: src/ParallelEnv.f90 [hunk 3]
```diff
@@ -739,6 +755,46 @@ SUBROUTINE scatter_SLK1_MPI_Env_type(myPE,sendbuf,recvbuf,root)
...
+!> @brief
+    SUBROUTINE bcast_SNK1_MPI_Env_type(myPE,buf,root)
+      CHARACTER(LEN=*),PARAMETER :: myName='bcast_SNK1_MPI_Env_type'
+      CLASS(MPI_EnvType),INTENT(INOUT) :: myPE
+      INTEGER(SNK),INTENT(IN) :: buf(:)
+      INTEGER(SIK),INTENT(IN),OPTIONAL :: root
+      INTEGER(SIK) :: rank
+      rank=0
+      IF(PRESENT(root)) rank=root
+      IF(0 <= rank .AND. rank <= myPE%nproc-1) THEN
```
Comment (bscollin) [inline src/ParallelEnv.f90]: Same

File: src/ParallelEnv.f90 [hunk 4]
```diff
@@ -759,6 +815,46 @@ SUBROUTINE bcast_SLK0_MPI_Env_type(myPE,buf,root)
...
+!> @brief
+    SUBROUTINE bcast_SLK1_MPI_Env_type(myPE,buf,root)
+      CHARACTER(LEN=*),PARAMETER :: myName='bcast_SLK1_MPI_Env_type'
+      CLASS(MPI_EnvType),INTENT(INOUT) :: myPE
+      INTEGER(SLK),INTENT(IN) :: buf(:)
+      INTEGER(SIK),INTENT(IN),OPTIONAL :: root
+      INTEGER(SIK) :: rank
+      rank=0
+      IF(PRESENT(root)) rank=root
+      IF(0 <= rank .AND. rank <= myPE%nproc-1) THEN
```
Comment (bscollin) [inline src/ParallelEnv.f90]: Same

File: src/ParallelEnv.f90 [hunk 5]
```diff
@@ -759,6 +815,46 @@ SUBROUTINE bcast_SLK0_MPI_Env_type(myPE,buf,root)
...
+!> @brief
+    SUBROUTINE bcast_SSK1_MPI_Env_type(myPE,buf,root)
+      CHARACTER(LEN=*),PARAMETER :: myName='bcast_SSK1_MPI_Env_type'
+      CLASS(MPI_EnvType),INTENT(INOUT) :: myPE
+      REAL(SSK),INTENT(IN) :: buf(:)
+      INTEGER(SIK),INTENT(IN),OPTIONAL :: root
+      INTEGER(SIK) :: rank
+      rank=0
+      IF(PRESENT(root)) rank=root
+      IF(0 <= rank .AND. rank <= myPE%nproc-1) THEN
```
Comment (bscollin) [inline src/ParallelEnv.f90]: ditto

File: src/ParallelEnv.f90 (hunk 1 above)
Comment (aarograh) [reply src/ParallelEnv.f90]: Oh... didn't mean to do this, but it's commented out in both versions, so it's not actually affecting anything.

**Comment by aarograh** [comment]
I just pushed a commit that adds DBC throughout ParallelEnv communication routines.  I also deleted a couple commented GENERIC statements since we're not using them.

**Comment by bscollin** [comment]
Looks like its failing without MPI.

**Pull request merged**